### ISSUE TITLE
Describe the difference between `add_points` and `add_mesh` in document

### ISSUE THREAD

**Issue by GuillemBarroso** [pyvista/pyvista#3696]
### Describe the feature you would like to be added.

I would like to have an ``add_planes`` method in the ``pyvista.Plotter`` object to add ``pyvista.Plane`` objects to the plotter.

I was able to use the ``add_planes_widget``, but I only want the plane itself and not the entire widget.

### Links to VTK Documentation, Examples, or Class Definitions.

NA

### Pseudocode or Screenshots

```python
import pyvista as pv
pl = pv.Plotter()
plane = pv.Plane(center, normal)
pl.add_planes(plane)
pl.show()
```

**Comment by tkoyama010** [comment]
Is [add_mesh](https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.add_mesh.html) what you want?
```python
import pyvista as pv
pl = pv.Plotter()
plane = pv.Plane()
pl.add_mesh(plane)
pl.show()
```

**Comment by GuillemBarroso** [comment]
Exactly! Thank you for the quick response.

However, I am not sure if the ``add_mesh`` docs are the most intuitive:
![image](https://user-images.githubusercontent.com/53979143/206398381-df09c9b8-a1c7-471f-bfaa-bca8c2c0b4d6.png)

It may just be me, but to me it was not really clear that ``add_mesh`` was also accepting ``pv.Plane`` objects.

I am closing this issue. Thanks again!

**Comment by AlejandroFernandezLuces** [comment]
It seems a bit counter intuitive to have `add_points` and `add_lines` operations, but not an `add_planes`. Maybe a simple wrapper for `add_mesh` can be implemented to have the `add_planes` function.

**Comment by tkoyama010** [comment]
@AlejandroFernandezLuces  Good point. My only concern is that the methods will grow indefinitely. For example, for Shere, we would need `add_sphere`. We need to address all objects under parametric_objects.py at least.

**Comment by MatthewFlamm** [comment]
My opinion is that we should not add these functions.  It would be never ending.  I would prefer to only have these helper functions that help cut through the mess that is the function signature of `add_mesh`.   For example, `add_points` forces the user to select one style that renders points.

https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.add_points.html

`add_plane` would have no such benefit that I can see, but I'm open to hear if there are benefits here.  A `pv.Plane` object is just a normal `DataSet`.

However, this issue highlights that we could do a better job in the documentation to describe the difference between `add_points` and `add_mesh`, i.e. `add_points` is a helper and only a thin wrapper around `add_mesh`.
